### DISCUSSION TITLE
Cannot read properties of undefined (reading 'keyword') using `@nx/enforce-module-boundaries`

### DISCUSSION THREAD

**Discussion by lisherwin** [nrwl/nx#20194] [General]
Hi!

I'm trying to introduce the eslint plugin rule `@nx/enforce-module-boundaries` to our monorepo to enforce module boundaries in our existing monorepo, but I'm getting this crash when running `yarn lint`.
```
[@lisherwin/rpc::eslint] 
[@lisherwin/rpc::eslint] Oops! Something went wrong! :(
[@lisherwin/rpc::eslint] 
[@lisherwin/rpc::eslint] ESLint: 8.53.0
[@lisherwin/rpc::eslint] 
[@lisherwin/rpc::eslint] TypeError: Error while loading rule '@nx/enforce-module-boundaries': Cannot read properties of undefined (reading 'keyword')
[@lisherwin/rpc::eslint] Occurred while linting /opt/actions-runner/_work/wallet/wallet/packages/rpc/src/autoConfirm.ts
[@lisherwin/rpc::eslint]     at ensureGlobalProjectGraph (/opt/actions-runner/_work/wallet/wallet/node_modules/@nx/eslint-plugin/src/utils/project-graph-utils.js:35:51)
[@lisherwin/rpc::eslint]     at readProjectGraph (/opt/actions-runner/_work/wallet/wallet/node_modules/@nx/eslint-plugin/src/utils/project-graph-utils.js:46:5)
[@lisherwin/rpc::eslint]     at create (/opt/actions-runner/_work/wallet/wallet/node_modules/@nx/eslint-plugin/src/rules/enforce-module-boundaries.js:133:137)
[@lisherwin/rpc::eslint]     at Object.create (/opt/actions-runner/_work/wallet/wallet/node_modules/@nx/eslint-plugin/node_modules/@typescript-eslint/utils/dist/eslint-utils/RuleCreator.js:38:20)
[@lisherwin/rpc::eslint]     at createRuleListeners (/opt/actions-runner/_work/wallet/wallet/node_modules/eslint/lib/linter/linter.js:910:21)
[@lisherwin/rpc::eslint]     at /opt/actions-runner/_work/wallet/wallet/node_modules/eslint/lib/linter/linter.js:1081:110
[@lisherwin/rpc::eslint]     at Array.forEach (<anonymous>)
[@lisherwin/rpc::eslint]     at runRules (/opt/actions-runner/_work/wallet/wallet/node_modules/eslint/lib/linter/linter.js:1018:34)
[@lisherwin/rpc::eslint]     at Linter._verifyWithoutProcessors (/opt/actions-runner/_work/wallet/wallet/node_modules/eslint/lib/linter/linter.js:1370:31)
[@lisherwin/rpc::eslint]     at Linter._verifyWithConfigArray (/opt/actions-runner/_work/wallet/wallet/node_modules/eslint/lib/linter/linter.js:1822:21)
```

This is the first time we're importing anything nx related into our app so I'm wondering if I'm missing some extra configuration. Looking at the source code, it seems like its trying to `readCachedProjectGraph()` before `createProjectGraphAsync()` is called. Any thoughts would be appreciated. Thanks in advance!

The changes are fairly minimal:
![Screenshot 2023-11-10 at 2 05 37 PM](https://github.com/nrwl/nx/assets/879552/1c037da1-0100-4a59-bdfd-3ffd83fa9fa1)

**Comment by rburgst** [comment]
I started having this problem after upgrading to nx from 16.10 to 17.1.3

**Comment by MSkrzypietz** [comment]
I started getting this error after upgrading to v17 as well (only in CI pipelines). The solution was to switch from executing the `eslint` cli to using `nx lint` with the `@nx/eslint:lint` executor.

**Comment by Robouste** [comment]
Same issue here when working with vscode. No warnings were showing anymore in the code and I had that same error in the eslint output.

It started working again after making a change to nx.json and then reverted it ¯\\\_(ツ)\_/¯

A colleague has the same issue but making changes to nx.json didn't make it work, so I have no clues why it started working again for me

**Comment by JonSilver** [comment]
It's all been a bit of a mess moving to v17, to be honest. So many issues. OK, we get some new ways of doing things... fine if they work, just a PITA if they don't. And they don't. Regretting ever "upgrading" from v16 at all.

**Comment by gergelyszerovay** [comment]
I had the same issue, and created a bug report: https://github.com/nrwl/nx/issues/20752
